internal/sdkv2provider/resource_cloudflare_teams_certificates.go

```diff
@@ -0,0 +1,150 @@
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+
+	if d.Get("gateway_managed").(bool) {
+		newTeamsCertificate := cloudflare.TeamsCertificateCreateRequest{
+			ValidityPeriodDays: d.Get("validity_period_days").(int),
+		}
+
+		tflog.Debug(ctx, fmt.Sprintf("Creating Cloudflare Teams Certificate from struct: %+v", newTeamsCertificate))
+
+		certificate, err := client.TeamsGenerateCertificate(ctx, accountID, newTeamsCertificate)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error creating Teams Certificate for account %q: %w", accountID, err))
+		}
+		d.SetId(certificate.ID)
+	}
+
+	if d.Get("activate").(bool) {
+		certificate, err := client.TeamsActivateCertificate(ctx, accountID, d.Get("id").(string))
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error activating Teams Certificate with id %q for account %q: %w", d.Get("id"), accountID, err))
+		}
+		d.Set("binding_status", certificate.BindingStatus)
+		d.Set("qs_pack_id", certificate.QsPackId)
+	}
+
+	return resourceCloudflareTeamsCertificateRead(ctx, d, meta)
+}
+
+func resourceCloudflareTeamsCertificateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+
+	if d.Get("activate").(bool) {
+		certificate, err := client.TeamsActivateCertificate(ctx, accountID, d.Get("id").(string))
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error activating Teams Certificate with id %q for account %q: %w", d.Get("id"), accountID, err))
+		}
+		d.Set("binding_status", certificate.BindingStatus)
+		d.Set("qs_pack_id", certificate.QsPackId)
+	} else {
+		certificate, err := client.TeamsDeactivateCertificate(ctx, accountID, d.Get("id").(string))
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("error deactivating Teams Certificate with id %q for account %q: %w", d.Get("id"), accountID, err))
+		}
+		d.Set("binding_status", certificate.BindingStatus)
+		d.Set("qs_pack_id", certificate.QsPackId)
+	}
+
+	return resourceCloudflareTeamsCertificateRead(ctx, d, meta)
+}
+
+func resourceCloudflareTeamsCertificateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+
+	Certificate, err := client.TeamsCertificate(ctx, accountID, d.Id())
+
+	if err != nil {
+		var notFoundError *cloudflare.NotFoundError
+		if errors.As(err, &notFoundError) {
+			tflog.Info(ctx, fmt.Sprintf("Teams Certificate %s no longer exists", d.Id()))
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error finding Teams Certificate %q: %w", d.Id(), err))
+	}
+
+	d.Set("id", Certificate.ID)
+	d.Set("enabled", Certificate.Enabled)
+	d.Set("binding_status", Certificate.BindingStatus)
+	d.Set("qs_pack_id", Certificate.QsPackId)
+	d.Set("type", Certificate.Type)
+	d.Set("uploaded_on", Certificate.UploadedOn.Format(time.RFC3339Nano))
```

Comment:
similar issue with these timestamps; they are all throwing nil pointer dereferences.

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareTeamsCertificate_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareTeamsCertificate_Basic
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x101757524]

goroutine 152 [running]:
github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider.resourceCloudflareTeamsCertificateRead({0x1020d2ab0, 0x140007d72d0}, 0x140007dce00, {0x1020b2380?, 0x14000b1e000})
	/Users/jacob/src/github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider/resource_cloudflare_teams_certificates.go:110 +0x4f4
github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider.resourceCloudflareTeamsCertificateCreate({0x1020d2ab0, 0x140007d72d0}, 0x140007dce00, {0x1020b2380, 0x14000b1e000})
	/Users/jacob/src/github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider/resource_cloudflare_teams_certificates.go:62 +0x500
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x140009d87e0, {0x1020d2a08, 0x140006d1050}, 0x140007dce00, {0x1020b2380, 0x14000b1e000})
	/Users/jacob/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:806 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140009d87e0, {0x1020d2a08, 0x140006d1050}, 0x14000fd6410, 0x140007dcc80, {0x1020b2380, 0x14000b1e000})
	/Users/jacob/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0x884
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000990138, {0x1020d2a08?, 0x140006d0fc0?}, 0x14000a809b0)
	/Users/jacob/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xaa4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14000f9a8c0, {0x1020d2a08?, 0x140006d0450?}, 0x140007d6000)
	/Users/jacob/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x2b4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x102076ca0, 0x14000f9a8c0}, {0x1020d2a08, 0x140006d0450}, 0x140007dc000, 0x0)
	/Users/jacob/go/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000acf600, {0x1020d2a08, 0x140006d03c0}, {0x1020d8940, 0x14000ae5c80}, 0x14000b42480, 0x1400090d620, 0x102c3f5f8, 0x0)
	/Users/jacob/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1369 +0xb58
google.golang.org/grpc.(*Server).handleStream(0x14000acf600, {0x1020d8940, 0x14000ae5c80}, 0x14000b42480)
	/Users/jacob/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1780 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/Users/jacob/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 162
	/Users/jacob/go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1030 +0x13c
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	1.721s
FAIL
make: *** [testacc] Error 1
```

are you able to run the acceptance tests locally and get these passing?

Reply:
Am now getting 
`=== RUN   TestAccCloudflareTeamsCertificate_Basic
    provider_test.go:82: [{0 You've attempted to add a new resource to the `terraform-plugin-sdkv2` which is no longer considered suitable for use. Due the number of known internal issues with `terraform-plugin-sdkv2` (most notably handling of zero values), we are no longer recommending using it and instead, advise using `terraform-plugin-framework` exclusively. If you must use terraform-plugin-sdkv2 for this new resource you should first discuss it with a maintainer to fully understand the impact and potential ramifications. Only then should you bump MAXIMUM_ALLOWED_SDKV2_RESOURCES to include your resource. []}]`
    
 Does this mean I should add the resource to /internal/framework/service, and not allow users of older TF versions to interact with certificates?

Reply:
initially, this should have been added as a framework based resource however, let's leave this one as is seeing how it's already got to this point. it doesn't restrict end users on what resources they can use; only which internals it relies on.

Reply:
even after bumping the number (which you can do here too), there is still an acceptance test failure.

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareTeamsCertificate_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareTeamsCertificate_Basic
    resource_cloudflare_teams_certificate_test.go:26: Step 1/2 error: Error running apply: exit status 1
        
        Error: error reading Teams Certificate type "": %!w(<nil>)
        
          with cloudflare_zero_trust_gateway_certificate.xzcchpxrwj,
          on terraform_plugin_test.tf line 12, in resource "cloudflare_zero_trust_gateway_certificate" "xzcchpxrwj":
          12: resource "cloudflare_zero_trust_gateway_certificate" "xzcchpxrwj" {
        
--- FAIL: TestAccCloudflareTeamsCertificate_Basic (1.11s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	2.069s
FAIL
make: *** [testacc] Error 1
```

Reply:
My bad, there's a bug on my api. Will reopen this PR when ready